packages/shared/src/components/notifications/InAppNotificationsTab.tsx

```diff
@@ -0,0 +1,267 @@
+import SquadModNotifications from './SquadModNotifications';
+import NotificationCheckbox from './NotificationCheckbox';
+import NotificationSwitch from './NotificationSwitch';
+
+const InAppNotificationsTab = (): ReactElement => {
+  const { logEvent } = useLogContext();
+  const { onTogglePermission } = usePushNotificationMutation();
+  const { isSubscribed, isInitialized } = usePushNotificationContext();
+  const { openModal } = useLazyModal();
+  const {
+    notificationSettings: ns,
+    toggleSetting,
+    toggleGroup,
+    getGroupStatus,
+  } = useNotificationSettings();
+
+  const onTogglePush = async () => {
+    logEvent({
+      event_name: isSubscribed
+        ? LogEvent.DisableNotification
+        : LogEvent.EnableNotification,
+      extra: JSON.stringify({
+        channel: NotificationChannel.Web,
+        category: NotificationCategory.Product,
+      }),
+    });
+    return onTogglePermission(NotificationPromptSource.NotificationsPage);
+  };
+
+  return (
+    <section className="flex flex-col gap-6 py-4">
+      <div className="flex flex-row justify-between gap-4 px-4">
+        <div className="flex flex-1 flex-col gap-1">
+          <Typography type={TypographyType.Body} bold>
+            Push notifications
+          </Typography>
+          <Typography
+            color={TypographyColor.Tertiary}
+            type={TypographyType.Footnote}
+          >
+            Turn this on to get real-time updates on your device. You’ll still
+            see in-app notifications even if this is off. Requires additional
+            device permissions.
+          </Typography>
+        </div>
+        <Switch
+          data-testid="push_notification-switch"
+          inputId="push_notification-switch"
+          name="push_notification"
+          className="w-20 justify-end"
+          compact={false}
+          checked={isSubscribed}
+          onToggle={onTogglePush}
+          disabled={!isInitialized}
+        />
+      </div>
+      <div className="flex flex-row justify-between gap-4 px-4">
+        <div className="flex flex-1 flex-col gap-1">
+          <Typography type={TypographyType.Body} bold>
+            Squad notifications
+          </Typography>
+          <Typography
+            color={TypographyColor.Tertiary}
+            type={TypographyType.Footnote}
+          >
+            Control notifications for new posts in squads you’ve joined. Scroll
+            down for moderation notification settings.
+          </Typography>
+        </div>
+        <Button
+          icon={<ArrowIcon className="m-auto rotate-90" />}
+          variant={ButtonVariant.Option}
+          onClick={() =>
+            openModal({ type: LazyModal.SquadNotificationSettings })
+          }
+        />
+      </div>
+      <NotificationSection>
+        <Typography type={TypographyType.Body} bold>
+          Activity
+        </Typography>
+        <NotificationContainer>
+          {ACTIVITY_NOTIFICATIONS.map((item) => (
```

Comment:
It's fine, but think this and the email one could share a render function where you simply pass the list.
This is already easier to follow though.